src/app/donation-start/donation-start-form/donation-start-form.component.ts

```diff
@@ -1330,79 +1316,6 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
     this.donationCreateError = true;
     this.stepper.previous(); // Go back to step 1 to surface the internal error.
   }
-
-  private preparePaymentRequestButton = (donation: Donation, paymentGroup: FormGroup) => {
-    if (this.skipPRBs) {
-      return;
-    }
-
-    if (this.paymentRequestButton) {
-      this.paymentRequestButton.clear();
-    }
-
-    const paymentRequestResultObserver: Observer<{billingDetails: PaymentMethod.BillingDetails | undefined, walletName: string}> = {
-      next: (observed) => {
-        if (observed.billingDetails && donation) {
-          this.matomoTracker.trackEvent(
-            'donate',
-            'stripe_prb_setup_success',
-            `Stripe PRB success for donation ${donation.donationId} to campaign ${this.campaignId}`,
-          );
-
-          // Set form and `donation` billing fields from PRB card's data.
-          paymentGroup.patchValue({
-            billingCountry: observed.billingDetails.address?.country,
-            billingPostcode: observed.billingDetails.address?.postal_code,
-          });
-
-          this.stripePaymentMethodReady = true;
-          this.stripePRBMethodReady = true;
-          this.removeStripeCardBillingValidators();
-          this.jumpToStep('Receive updates');
-
-          return;
-        }
-        // Else there was no Payment Method (or details), so the PRB failed.
-
-        this.stripePaymentMethodReady = false;
-        this.stripePRBMethodReady = false;
-        this.addStripeCardBillingValidators();
-
-        // I *think* `payWithStripe()` also sets `this.stripeError` & `this.stripeResponseErrorCode`,
-        // but that this event's handling typically happens later. So we can take the opportunity to replace
-        // `stripeError` with a more specific, wallet-targeted explanation.
-        if (this.stripeResponseErrorCode === 'card_declined') {
-          const walletFriendlyName = observed.walletName === 'apple_pay' ? 'Apple Pay' : 'Google Pay';
-          this.stripeError = `Payment failed. Please check your card's billing address in your ${walletFriendlyName} wallet matches your the address your bank has, then try again.`;
-        }
-      },
-
-      error: (_) => {
-        this.stripePaymentMethodReady = false;
-        this.stripePRBMethodReady = false;
-        this.addStripeCardBillingValidators();
-        this.stripeError = 'Payment method handling failed';
-        this.stripeResponseErrorCode = undefined;
-      },
-      complete: () => {},
-    };
-
-    if (! this.stripeElements) {
-      throw new Error("Stripe elements not ready");
-    }
-
-    this.paymentRequestButton = this.stripeService.getPaymentRequestButton(donation, this.stripeElements, paymentRequestResultObserver);
```

Comment:
:+1: . It would be great to have a static analysis check to say we're don't have dead code, unless either we've stated that its there for some purpose that the SA tool can't detect, like use by code we intend to write in future, or dynamic use.

Done in 0fbe33dd